rest/zone_test.go

```diff
@@ -9,80 +11,148 @@ import (
 		"example.org",
 		"example.io",
 	}
-
-	assert.Equal(t, 4, len(zones))
+	assert.Equal(t, 4, len(zones), "found wrong number of zones")
 	for idx := range zones {
-		assert.Equal(t, expected[idx], zones[idx].Zone)
+		assert.Equal(t, expected[idx], zones[idx].Zone, "zone.Zone does not match")
 	}
 }
 
-func TestPagination_zoneRecords(t *testing.T) {
+func TestZone_get_pagination(t *testing.T) {
 	// It should follow Links and gather all Records into the Zone
-	testServer := mockAPI(t)
+	testServer := mockAPI(t, responseMap{"/zones/coolzone.cool": handleZoneGet})
 	doer := testServer.Client()
 	client := NewClient(doer, SetEndpoint(testServer.URL))
 
-	zone, _, _ := client.Zones.Get("coolzone.cool")
+	zone, resp, err := client.Zones.Get("coolzone.cool")
+	assert.NoError(t, err, "expected no error")
+	assert.Equal(t, 200, resp.StatusCode, "status code does not match")
 
 	expected := []string{
 		"a.coolzone.cool",
 		"b.coolzone.cool",
 		"c.coolzone.cool",
 		"d.coolzone.cool",
 	}
-
-	assert.Equal(t, "coolzone.cool", zone.Zone)
-	assert.Equal(t, 4, len(zone.Records))
+	assert.Equal(t, "coolzone.cool", zone.Zone, "zone.Zone does not match")
+	assert.Equal(t, 4, len(zone.Records), "found wrong number of records")
 	for idx := range zone.Records {
 		assert.Equal(t, expected[idx],  zone.Records[idx].Domain)
 	}
+
 }
 
-func TestPagination_disabled(t *testing.T) {
+func TestZone_pagination_disabled(t *testing.T) {
 	// It should not follow Links
-	testServer := mockAPI(t)
+	testServer := mockAPI(t, responseMap{"/zones": handleZoneList, "/zones/coolzone.cool": handleZoneGet})
 	doer := testServer.Client()
 	client := NewClient(doer, SetEndpoint(testServer.URL), SetFollowPagination(false))
 
-	zones, _, _ := client.Zones.List()
+	zones, resp, err := client.Zones.List()
+	assert.NoError(t, err, "expected no error")
+	assert.Equal(t, 200, resp.StatusCode, "status code does not match")
+
 	expected := []string{
 		"example.com",
 		"coolzone.cool",
 	}
 	assert.Equal(t, 2, len(zones))
 	for idx := range zones {
-		assert.Equal(t, expected[idx], zones[idx].Zone)
+		assert.Equal(t, expected[idx], zones[idx].Zone, "zone.Zone does not match")
 	}
 
-	zone, _, _ := client.Zones.Get("coolzone.cool")
+	zone, resp, err := client.Zones.Get("coolzone.cool")
+	assert.NoError(t, err, "expected no error")
+	assert.Equal(t, 200, resp.StatusCode, "status code does not match")
+
 	expected = []string{
 		"a.coolzone.cool",
 		"b.coolzone.cool",
 	}
-	assert.Equal(t, "coolzone.cool", zone.Zone)
-	assert.Equal(t, 2, len(zone.Records))
+	assert.Equal(t, "coolzone.cool", zone.Zone, "zone.Zone does not match")
+	assert.Equal(t, 2, len(zone.Records), "found wrong number of records")
 	for idx := range zone.Records {
-		assert.Equal(t, expected[idx],  zone.Records[idx].Domain)
+		assert.Equal(
+			t, expected[idx],  zone.Records[idx].Domain, "record.Domain does not match",
+		)
 	}
 }
 
-func mockAPI(t *testing.T) *httptest.Server {
+func TestZone_http_errors(t *testing.T) {
```

Comment:
Should be split into two functions and named something like `TestZone_ListHTTPErrors` and `TestZone_ListHTTPErrors`